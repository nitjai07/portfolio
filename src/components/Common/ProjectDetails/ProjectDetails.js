import React, { Component } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { expressStaticUrl } from '../../../utils/staticSideNavData';
import Heading from '../Heading/Heading';
// import Typewriter from '../Typewriter/Typewriter';
// import mainImg from '../../../assets/images/digi_5.jpg';

class ProjectDetails extends Component {
  state = {
    // imageSrc: pMainImg,
    // changeImgStatus: false
  };

  componentWillMount() {
    window.scrollTo(0, 0);
  }

  render() {
    return (
      <ReactCSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={400}
        transitionEnterTimeout={400}
        transitionLeaveTimeout={100}
        transitionName="loadComponent"
      >
        <div className="projectwrapper">
          <div className="projectwrapper__topsection">
            <Link to="/projects" className="projectwrapper__topsection__back">
              <FontAwesomeIcon icon="arrow-left" />
              <span>back to projects</span>
            </Link>
            <Heading headingText={this.props.projectData.heading} />
            <p className="projectDesc">{this.props.projectData.shortDesc}</p>
            <div className="projectwrapper__topsection__moredetails">
              <ul>
                {this.props.projectData.client && (
                  <li>
                    <h3>Client</h3>
                    <p>{this.props.projectData.client}</p>
                  </li>
                )}
                <li>
                  <h3>Role</h3>
                  <p>{this.props.projectData.role}</p>
                </li>
                <li>
                  <h3>Date</h3>
                  <p>{this.props.projectData.date}</p>
                </li>
                <li>
                  <h3>Tech</h3>
                  <p>{this.props.projectData.tech}</p>
                </li>
              </ul>
            </div>
            {this.props.projectData.websiteUrl && (
              <div className="projectwrapper__topsection__visitsite">
                <a href={this.props.projectData.websiteUrl} className="primary-btn" target="_blank">
                  Visit The Website
                </a>
              </div>
            )}
          </div>
          <div className="projectwrapper__images">
            {this.props.projectData.images.map(image => (
              <img key={image} src={`${expressStaticUrl.url}${image}`} alt="" />
            ))}
          </div>
          <div className="projectwrapper__next-prev-projects">
            <Link to={`/projects/${this.props.prevProj}`} className="">
              <FontAwesomeIcon icon="arrow-left" />
              <span> Previous Project</span>
            </Link>
            <Link to={`/projects/${this.props.nextProj}`} className="">
              <span>Next Project </span>
              <FontAwesomeIcon icon="arrow-right" />
            </Link>
          </div>
        </div>
      </ReactCSSTransitionGroup>
    );
  }
}

ProjectDetails.propTypes = {
  projectData: PropTypes.object.isRequired,
  prevProj: PropTypes.string.isRequired,
  nextProj: PropTypes.string.isRequired
};

// ProjectDetails.defaultProps = {
//   history: ''
// };

export default withRouter(ProjectDetails);
