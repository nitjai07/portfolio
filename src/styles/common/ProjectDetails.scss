
.projectwrapper {
  &__topsection {
    min-height: 500px;
    .commonheading {
      width: 95%;
      margin-top: -20px;
      text-align: center;
      padding-top: 0;
      -webkit-mask-image: linear-gradient(-75deg, rgba(0,0,0,.6) 30%, $tertiaryColor3 50%, rgba(0,0,0,.6) 70%);
      -webkit-mask-size: 200%;
      animation: shine 2s infinite;
      @include mqueryphone(320px, 767px) {
        margin-top: 30px;
      }
      &:before, &:after {
        content: "",
      }    
    }
    .projectDesc {
      margin: 3% auto 3% auto;
      display: inline-block;
      width: 95%;
      text-align: center;
      color: $tertiaryColor1;
      font-size: 24px;
      box-shadow: 0px 0px 20px 5px $primaryColor3;
      padding: 20px;
      box-sizing: border-box;
    }
    &__back {
      position: fixed;
      top: 4%;
      left: 6%;
      color: $tertiaryColor1;
      animation: floatHorizontal 2s linear 0s infinite;
      font-size: 30px;
      text-decoration: none;
      transition: all 1s;
      z-index: 1;
      @include mqueryipad(768px, 1024px) {
        top: 5%;
        left: 12%;
      }
      @include mqueryphone(320px, 767px) {
        top: 3%;
        left: 50%;
      }
      svg {
        transition: all 1s;
        opacity: 1;
      }
      &:hover {
        svg {
          font-size: 0;
          opacity: 0;
        }
        span {
          font-size:14px;
          opacity: 1;
        }
      }
      span {
        font-size: 0;
        opacity: 0;
        transition: all 1s;
      }
    }
    &__moredetails {
      ul {
        width: 95%;
        @include mqueryphone(320px, 767px) {
          max-width: 85%;
        }
        li {
          display: inline-block;
          list-style-type: none;
          width: 33%;
          @include mqueryphone(320px, 767px) {
            vertical-align: top;
          }
          >p {
            color: $secondaryColor2;
            margin-top: 10px;
          }
          >h3 {
            color: $tertiaryColor1;
            font-weight: 700;
            margin-bottom: 0;
            font-family: 'Montserrat', sans-serif;
          }
        }
      }
    }
    &__visitsite {
      text-align: center;
      width: 95%;
      >a {
        margin-top: 0;
        margin-bottom: 30px;
        &:hover {
          svg {
            color: $tertiaryColor3;
          }
        }
        svg {
          font-size: 18px;
          color: $primaryColor2;
          margin-right: 5px;
        }
      }
    }
  }
  &__images {
    width: 95%;
    text-align: center;
    >img {
      max-width: 100%;
      display: block;
      margin: 0 auto 100px auto;
      animation: floatVertical 2s linear 0s infinite;
      max-height: 500px;
      &:first-of-type {
        margin-top: 50px;
      }
      &:last-of-type {
        margin-bottom: 100px;
      }
    }
  }

  &__next-prev-projects {
    text-align: center;
    width: 95%;
    margin-top: 50px;
    font-size: 24px;
    background: rgba(0,0,0,0.2);
    padding: 20px;
    box-sizing: border-box;
    >a {
      color: $tertiaryColor1;
      text-decoration: none;
      &:first-of-type {
        margin-right: 10%;
        svg {
          margin-right: 15px;
        }
      }
      &:hover {
        svg {
          animation: floatHorizontal 2s linear 0s infinite;
        }
      }
      svg {
        vertical-align: middle;
        margin-left: 15px;
      }
    }
    @include mqueryphone(320px, 767px) {
      font-size: 14px;
      >a {
        &:first-of-type {
          margin-right: 5%;
        }
      }
    }
  }
}