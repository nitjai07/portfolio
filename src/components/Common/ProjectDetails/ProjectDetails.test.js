import React from 'react';
import { shallow } from 'enzyme';
import ProjectDetails from './ProjectDetails';
import meImg from '../../../assets/images/me_1.jpg';

const ProjectData = {
  DnD: {
    heading: 'Deal or No Deal',
    shortDesc: 'Deal or no deal is a fictional game based on the inidian TV Game show.',
    client: 'Training MFRP',
    role: 'Front End Developer',
    date: 'April 2015',
    images: [meImg],
    tech: 'HTML5, CSS3, JavaScript'
  }
};

test('Project Details Component renders correctly', () => {
  const component = shallow(
    <ProjectDetails ProjectData={ProjectData} prevProj="previous project name" nextProj="next project name" />
  );
  expect(component).toMatchSnapshot();
});
