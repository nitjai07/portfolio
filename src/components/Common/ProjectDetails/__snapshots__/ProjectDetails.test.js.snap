// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Details Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Route>
    [Function]
</Route>,
  "nodes": Array [
    <Route>
      [Function]
</Route>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <withRouter(ProjectDetails)
      ProjectData={
            Object {
                  "DnD": Object {
                    "client": "Training MFRP",
                    "date": "April 2015",
                    "heading": "Deal or No Deal",
                    "images": Array [
                      "me_1.jpg",
                    ],
                    "role": "Front End Developer",
                    "shortDesc": "Deal or no deal is a fictional game based on the inidian TV Game show.",
                    "tech": "HTML5, CSS3, JavaScript",
                  },
                }
      }
      nextProj="next project name"
      prevProj="previous project name"
/>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <Route>
      [Function]
</Route>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <withRouter(ProjectDetails)
    ProjectData={
        Object {
            "DnD": Object {
              "client": "Training MFRP",
              "date": "April 2015",
              "heading": "Deal or No Deal",
              "images": Array [
                "me_1.jpg",
              ],
              "role": "Front End Developer",
              "shortDesc": "Deal or no deal is a fictional game based on the inidian TV Game show.",
              "tech": "HTML5, CSS3, JavaScript",
            },
          }
    }
    nextProj="next project name"
    prevProj="previous project name"
/>,
}
`;
